src/huggingface_hub/inference/_common.py

```diff
@@ -355,7 +355,7 @@ def _format_chat_completion_stream_output(
 
 async def _async_yield_from(client: "ClientSession", response: "ClientResponse") -> AsyncIterable[bytes]:
     async for byte_payload in response.content:
-        yield byte_payload
+        yield byte_payload.strip()
```

Comment:
@Wauplin Btw, the two earlier `byte_payload.rstrip()` may not be necessary, as `_format_text_generation_stream_output` and `_format_chat_completion_stream_output` receive "[DONE]" w/o "\n" from the regular client (which uses `.iter_lines` and strips newlines naturally), while the `byte_payload` for the async client passes through `_async_yield_from` here first.

Maybe just a micro optimization, in case performance ever becomes an issue here.

Reply:
Yes, I saw that but better to be safer. Each method taken individually should be able to work with unstripped data and return stripped one. Usually inference time is much higher than the parsing overhead.